File: Wifi Range [19 - April - 2023]/Approach 1(Brute_Force).cpp
//   >> C++ CODE <<

bool wifiRange(int N, string S, int X){
        
        vector<bool> visited(N, false);
        for (int i = 0; i < N; i++) {
            if (S[i] == '1') {
                for (int j = max(0, i - X); j <= min(i + X, N - 1); j++) {
                    visited[j] = true;
                }
            }
        }
        for (bool val : visited) {
            if (val == false) {
                return false;
            }
        }
        return true;

        
    }

File: Wifi Range [19 - April - 2023]/Approach 1(Brute_Force).java
//   >> JAVA CODE <<


boolean wifiRange(int N, String S, int X) 
    { 
        boolean[] visited = new boolean[N];
        Arrays.fill(visited, false);
        for (int i = 0; i < N; i++) {
            if (S.charAt(i) == '1') {
                for (int j = Math.max(0, i - X); j <= Math.min(i + X, N - 1); j++) {
                    visited[j] = true;
                }
            }
        }
        for (boolean val : visited) {
            if (val == false) {
                return false;
            }
        }
        return true;
    }